projects/js-packages/partner-coupon/components/redeem-partner-coupon-post-connection/test/component.jsx

```diff
@@ -0,0 +1,198 @@
+
+		const logo = screen.getByAltText(
+			'Logo of Company name who are offering a coupon in partnership with Jetpack'
+		);
+		expect( logo );
+		expect( logo.width ).equals( 150 );
+		expect( logo.height ).equals( 100 );
+	} );
+
+	it( 'do not try to show partner logo if we do not have any', () => {
+		const partnerCouponCopyWithoutLogo = partnerCoupon;
+		delete partnerCouponCopyWithoutLogo.partner.logo;
+
+		const props = {
+			...requiredProps,
+			partnerCoupon: partnerCouponCopyWithoutLogo,
+		};
+
+		const { container } = render( <RedeemPartnerCouponPostConnection { ...props } /> );
+
+		// We use querySelector because using typical screen.* selectors will give an
+		// error if the component doesn't exist; and we specifically want to ensure
+		// it doesn't exist while displaying the "Set up button".
+		expect(
+			container.querySelector(
+				'img[alt="Logo of Company name who are offering a coupon in partnership with Jetpack"]'
+			)
+		).to.not.exist;
+	} );
+
+	it( 'description is shown with product name', () => {
+		render( <RedeemPartnerCouponPostConnection { ...requiredProps } /> );
+
+		expect(
+			screen.getAllByText(
+				'Redeem your coupon and get started with Awesome Product for free the first year! Never worry about losing your data, ever.'
+			)
+		).to.exist;
+	} );
+
+	it( 'shows product features', () => {
+		render( <RedeemPartnerCouponPostConnection { ...requiredProps } /> );
+
+		// eslint-disable-next-line no-unused-vars
+		for ( const [ key, feature ] of Object.entries( partnerCoupon.product.features ) ) {
+			expect( screen.getAllByText( feature ) ).to.exist;
+		}
+	} );
+
+	it( 'redeem button redirects with all expected parameters', () => {
+		render( <RedeemPartnerCouponPostConnection { ...requiredProps } /> );
+
+		const redeemButton = screen.getByRole( 'button', {
+			name: 'Redeem Awesome Product',
+		} );
+		expect( redeemButton ).to.exist;
+		fireEvent.click( redeemButton );
+
+		// Make sure we only redirect once, and it's with the same value as getRedirectUrl.
+		expect( locationAssignSpy.calledOnce );
+		expect(
+			locationAssignSpy.withArgs(
+				sinon.match.same(
+					getRedirectUrl( 'jetpack-plugin-partner-coupon-checkout', {
+						path: 'awesome-product',
+						site: 'example.com',
+						query: 'coupon=TEST_TST_1234',
+					} )
+				)
+			).calledOnce
+		);
+
+		// Make sure we call track before calling location.assign.
+		expect( locationAssignSpy.calledAfter( recordEventStub ) );
+	} );
+
+	it( 'redeem button redirects after tracking event', () => {
+		render( <RedeemPartnerCouponPostConnection { ...requiredProps } /> );
+
+		const redeemButton = screen.getByRole( 'button', {
+			name: 'Redeem Awesome Product',
+		} );
+		expect( redeemButton ).to.exist;
+		fireEvent.click( redeemButton );
+		expect( locationAssignSpy.calledOnce );
+
+		// Make sure we trigger tracking event before redirecting.
+		expect( locationAssignSpy.calledAfter( recordEventStub ) );
+	} );
```

Comment:
This seems to be covered by the previous test or am I misunderstanding what it's doing? If the goal is to isolate the tracking behavior into its own test so we don't test for multiple things in the previous test then it makes some sense but we're still testing twice for it.

Reply:
It's purely to separate it out to its own test. So both `redeem button redirects with all expected parameters` and `redeem button redirects after tracking event` finds the redeem button and triggers a click but:

* `redeem button redirects with all expected parameters`: makes sure we pass along expected parameters (so compares with "getRedirectUrl")
* `redeem button redirects after tracking event`: makes sure the tracking event is triggering before the redirect

Technically we can combine them again and name the test something like `redeem button correctly redirect to checkout` but the current specificity seems better to be - but I don't have a strong opinion about it, so I'm more than willing to change it 🙈

Reply:
I have no strong opinion either to be honest, I'm fine with a test testing 2 things when it's a simple one-liner added /shrug.

Reply:
I'll keep it as it then, just because I've previously seen tests be refactored where some conditions were lost because they weren't part of the test description 🤷  😄